core/src/components/input/input.tsx

```diff
@@ -836,6 +857,30 @@ export class Input implements ComponentInterface {
                 <ion-icon aria-hidden="true" icon={mode === 'ios' ? closeCircle : closeSharp}></ion-icon>
               </button>
             )}
+            {this.showPasswordToggle && !readonly && !disabled && (
```

Comment:
The design document described rendering this content as the default content of the `end` slot. 

Did we experience an issue or additional consideration to not render it as the default slot content?

Reply:
Actually, this is something that I overlooked in the design document. But now testing it as the default content of the end slot, I noticed that the password toggle button will be hidden when the end slot is used. 

I think this would impact some use-cases. For example, if the developer wants to use the password toggle and also add a green checkmark icon to the end slot to represent the validation of the strength of the password, it will not be possible to achieve.

Please let me know your thoughts on it.

Reply:
> But now testing it as the default content of the end slot, I noticed that the password toggle button will be hidden when the end slot is used.

This is an intentional behavior. If you place the toggle password content in the end slot that's considered the "default content". Developers can override the default behavior to add their own logic (such as a custom password toggle button).

> I think this would impact some use-cases. For example, if the developer wants to use the password toggle and also add a green checkmark icon to the end slot to represent the validation of the strength of the password, it will not be possible to achieve.

This was [discussed](https://outsystemsrd.atlassian.net/wiki/spaces/RDMBVS/pages/3939729499?focusedCommentId=3990618567#comment-3990618567) in the design document. The concern from design was that you could potentially have 4 icons on the same line which will reduce the visible viewport for the input text. For example, you could have a start icon, an end icon, a clear input icon, and a validation icon all on the same line.

The idea here is that by using the end slot we can leverage the existing infrastructure and avoid the problem altogether.

Reply:
Let's talk about this internally.